certbot/certbot/util.py

```diff
@@ -61,6 +61,19 @@
 _LOCKS = OrderedDict() # type: OrderedDict[str, lock.LockFile]
 
 
+def env_no_snap_for_external_calls():
+    """
+    Returns a modified env to pass to Popen
+    """
+    env = os.environ.copy()
+    if 'SNAP' not in env:
+        return env
+    for path_name in ('PATH', 'LD_LIBRARY_PATH', 'PYTHONPATH'):
+        if path_name in env:
+            env[path_name] = ':'.join(x for x in env[path_name].split(':') if env['SNAP'] not in x)
```

Comment:
I don't think this filters out the DNS plugin snap paths from `PYTHONPATH` since `SNAP` refers to the Certbot snap right?

Reply:
Oh hm that's a good point. One way we could address this is to run the same thing as we do in `certbot.wrapper` and remove the things generated by that. Another thing we could do, since we're modifying internal certbot code to deal with packaging issues anyway, is to go back to setting a specific path for finding plugins. I don't love the thought of doing heuristic things like looking for the word "snap" on its own; the fact that the `SNAP` env var is explicitly going to be set in relevant cases makes me much more comfortable doing this. Thoughts?

Reply:
I don't have that strong of an opinion, but I honestly think I prefer setting a Certbot specific variable for finding plugins. That code is already written (at least somewhere in `git` history) and it seems to me like the simplest and most robust approach I can think of.